src/shared/GlobalTopBanners/TokenlessBanner/TokenlessBanner.tsx

```diff
@@ -0,0 +1,160 @@
+import isString from 'lodash/isString'
+import { useState } from 'react'
+import { useParams } from 'react-router'
+
+import { useOrgUploadToken } from 'services/orgUploadToken'
+import { useOwner } from 'services/user'
+import { useFlags } from 'shared/featureFlags'
+import A from 'ui/A'
+import Button from 'ui/Button'
+import { CopyClipboard } from 'ui/CopyClipboard'
+import Icon from 'ui/Icon'
+import { Tooltip } from 'ui/Tooltip'
+import TopBanner from 'ui/TopBanner'
+
+interface UseParams {
+  owner: string
+}
+
+interface UseOrgUploadTokenParams {
+  provider: string
+  owner: string
+}
+
+function OrgUploadTokenTooltip({ orgUploadToken }: { orgUploadToken: string }) {
+  const [isTokenVisible, setIsTokenVisible] = useState(true)
+  const encodedToken = orgUploadToken.replace(/\w|[^-]/g, 'x')
```

Comment:
This should probably be encoded at the time of calling the hook right? Otherwise it's still available in the network tab

If the purpose is to just blankly show and hide the token with a string then I think we can use a variable with the const "xxx-xxx" to mimic this functionality without spending cycles doing the replace

Reply:
Nope, we don’t need to go that far. This is purely for accessibility, such as when someone is nearby and you want to ensure it's only visible to you. I think replacing it with the exact number of characters makes more sense, as that's how most 'hide password' functionality typically works. It's mimicking the functionality of the global upload token settings tab.

Reply:
Fair enough, it's not too expensive either way